### ISSUE TITLE
how does one override the default DPI value for graphics -> png ?

### ISSUE THREAD

**Issue by gpetty** [brucemiller/LaTeXML#651]
I have been unable to track down any online information relating to the setting of the DPI value for the conversion of included figures from PDF to PNG in latexmlpost.  Browsing the Graphics.pm file, it appears that it should be possible to set an option that overrides the value of 92 that is assumed by default but, not being a Perl programmer, I can't tell where or how that option would be supplied to latexmlpost.

Basically, the figures currently come out looking pretty poor resolution, and I'd like to roughly double the resolution without manually editing the Perl source code. 

As an aside, I'm wondering whether there could be an option to specify the longest image dimension in pixels rather than the DPI value, since some PDF figures are "natively" rather small but get scaled in a LaTeX document to the desired physical width.

**Comment by brucemiller** [comment]
Hmm, there's a magnification option for math images, but not an option for the dpi conversion more generally.  But why aren't you using the width option to includegraphics?

**Comment by gpetty** [comment]
On Sat, Aug 22, 2015 at 8:03 AM, bruce miller <notifications@github.com>
wrote:

> Hmm, there's a magnification option for math images, but not an option for
> the dpi conversion more generally. But why aren't you using the width
> option to includegraphics?
>
>
Sorry I'm answering this more than a month after you wrote it -- it had
landed in a folder I don't usually monitor.

I do use the width option to set the size of the PDF images to be
commensurate with the width of the text on the printed page.  As I noted,
many turn out fairly coarsely rasterized.

Does the dpi conversion look at the native width of the PDF image or the
width after scaling?

 - Grant

**Comment by dginev** [comment]
Let me piggyback on this issue.

While I don't care about DPI myself, I care a lot about file size. I just had ImageMagick, through LaTeXML, generate a 8.5 **GB** file while converting an unknown arXiv article. I would like to set a hard maximum filesize to images generated by LaTeXML, and could think that 50MB should be enough for any arXMLiv-related task.

While I do thing that exposing **all** of ImageMagick's options from LaTeXML's command line is terrible idea (there are tens if not hundreds of them), either a meaningful default, or a postprocessor parameter would go a long way.

**Comment by dginev** [comment]
I understand the rasterization problems a bit better now (from work experience) and indeed ImageMagick is often suboptimal in converting. 

That said, there is no silver bullet for ideal image conversion, and I have spent over a week searching for one. So the best advice here would be to convert to web images that are satisfactory outside of latexml, and include those directly via includegraphics. As much as I would wish latexml could do perfect conversions of images at all times, I am pretty confident that is unrealistic, even with exposing all of imagemagick's options - sometimes imagemagick itself is not the perfect tool for the job, as using ghostscript as a delegate for a vectorial PDF image can have really disappointing results.

So closing with "wontfix" here, apologies -- can reopen if we get some great automation ideas. For now can only advise to use a separate advanced tool for image conversion. Bruce can overrule me of course, but I really think this is a lot more difficult to do right than adding a single option.

**Comment by asmaier** [comment]
Why not offer a simple magnification option as for math images for all images? Wouldn't that be enough to help with this issue?
